pkg/loop/internal/pb/chain_reader.proto

```diff
@@ -151,7 +151,7 @@ message Comparator  {
 }
 
 message Block{
-  uint64 block_number = 1;
```

Comment:
Should I instead deprecate the previous field and add a new one because of [backward compatibility](https://protobuf.dev/programming-guides/dos-donts/#change-type)?

```
message Block {
  uint64 block_number_deprecated = 1 [deprecated = true]; // rename prev field and mark as deprecated
  string block_number = 3; // new field with string type that has the old name
  ComparisonOperator operator = 2;
}
```

Reply:
Usually that is the best practice, but I don't see anyone using block querying so as long as you update tests in core evm it is fine to make a breaking change.

Reply:
You'll have to update the PGDSLParser  [here](https://github.com/smartcontractkit/chainlink/blob/05ef7fdbb115f55a85bcbbc5402350818501e1f5/core/chains/evm/logpoller/parser.go#L45), and check if any of the LP tests need to be fixed after this change

Reply:
Fixed in this PR: https://github.com/smartcontractkit/chainlink/pull/14033